library/core/src/sync/atomic.rs

```diff
@@ -3420,7 +3420,7 @@ macro_rules! atomic_int {
             ///
             #[doc = concat!("let atomic = ", stringify!($atomic_type), "::new(1);")]
             ///
-            /// // SAFETY: Safe as long as `my_atomic_op` is atomic.
+            /// // SAFETY: Safe as long as `my_atomic_op` does not lead to a data race.
```

Comment:
The SAFETY comment is insufficient because the safety also depends on the pointer being valid/aligned/etc., but that is beyond the scope of this PR.

I think the comment should be "SAFETY: `my_atomic_op` only uses atomic operations so it will not lead to a data race."

There are other examples, such as the one that I gave in issue #136669, that demonstrate where a non-atomic operation may be used safely. Maybe we should add an example along those lines.